### PR TITLE
remove PipelineStatus and PipelineResourceStatus

### LINKED ISSUE #1636: api proposal: delete PipelineStatus
# Expected Behavior
`Pipeline` and `Task` would have similar API patterns.

# Actual Behavior
`Pipeline` has a corresponding `PipelineStatus` which is empty. The comments listed with it suggest that a Pipeline can't have a status:
```
// PipelineStatus does not contain anything because Pipelines on their own
// do not have a status, they just hold data which is later used by a
// PipelineRun.
```

`Task` is generated with `+genclient:noStatus`.

I would propose we delete `PipelineStatus`. This may be a breaking change for a user who has decided to use `PipelineStatus` in their Go code, however I'm not sure what they would actually do with it, given its empty.

# Steps to Reproduce the Problem

1.
2.
3.

# Additional Info

A quick grep through github.com/tektoncd/pipeline seems to indicate the type is not used.

### PR DESCRIPTION
**Pull request by poy**
# Changes

This removes two struct types:

* `PipelineStatus`
* `PipelineResourceStatus`

These Status types were unused and did not follow the pattern of using
Knative's DuckType. Other types such as `Task` do not have a Status type
and instead defer to their corresponding `TaskRun` status.

fixes #1636

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```
Removes unused `status` fields from `Pipeline` and `PipelineResource`.
```

### CODE REVIEW

File: pkg/pod/entrypoint.go [hunk 1]
```diff
@@ -141,7 +141,7 @@ func orderContainers(entrypointImage string, steps []corev1.Container) (corev1.C
 func UpdateReady(kubeclient kubernetes.Interface, pod corev1.Pod) error {
 	newPod, err := kubeclient.CoreV1().Pods(pod.Namespace).Get(pod.Name, metav1.GetOptions{})
 	if err != nil {
-		return fmt.Errorf("Error getting Pod %q when updating ready annotation: %w", pod.Name, err)
+		return fmt.Errorf("Error getting Pod %q when updating ready annotation: %v", pod.Name, err)
```
Comment (dlorenc) [inline pkg/pod/entrypoint.go]: Were these changes intended to be part of this PR?

Comment (poy) [reply pkg/pod/entrypoint.go]: Well I'm not sure if it has something to do with go 1.13, but the complier was complaining about those.

Comment (poy) [reply pkg/pod/entrypoint.go]: I'll revert that change though.

**Comment by poy** [comment]
Are we not using go 1.13?

**Comment by vdemeester** [comment]
/hold
/lgtm
Let's wait for 0.9.0 to be released before pushing this change.

**Comment by vdemeester** [comment]
@poy can you squash your commits ? :angel:

**Comment by bobcatfish** [comment]
I updated the release notes in the PR desciription

**Comment by vdemeester** [comment]
@poy also needs a rebase :bowing_man:

**Comment by poy** [comment]
@vdemeester Rebased and squashed!

**Comment by bobcatfish** [comment]
@vdemeester sounds like you're already on this but i think this has some implications for CLI + Dashboard + and also Triggers (@dibyom ) - I just tried to use latest triggers with latest pipelines and got:

```
"Could not create resources for \"\": couldn't create resource with group version kind \"tekton.dev/v1alpha1, Resource=pipelineresources\": Internal error occurred: admission webhook \"webhook.tekton.dev\" denied the request: mutation failed: cannot decode incoming new object: json: unknown field \"status\""
```

**Comment by vdemeester** [comment]
@bobcatfish ah interesting… because there was no `omitempty`.. yeah… we need to bump the dependency to have the updated struct to have it work… that's not the best then because this will break user…

**Comment by vdemeester** [comment]
See #1762 for backward compat.

**Pull request merged**